.github/workflows/forward-compatibility.yaml

```diff
@@ -0,0 +1,131 @@
+# Copyright NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Forward Compatibility
+
+on:
+  schedule:
+    - cron: '0 2 * * 1'  # Weekly on Monday at 2 AM UTC
+  workflow_dispatch:
+    inputs:
+      toolkit_image:
+        description: 'Override container-toolkit image'
+        required: false
+        type: string
+      device_plugin_image:
+        description: 'Override device-plugin image'
+        required: false
+        type: string
+      mig_manager_image:
+        description: 'Override mig-manager image'
+        required: false
+        type: string
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  fetch-latest-images:
+    runs-on: ubuntu-latest
+    outputs:
+      toolkit_image: ${{ steps.images.outputs.toolkit_image }}
+      device_plugin_image: ${{ steps.images.outputs.device_plugin_image }}
+      mig_manager_image: ${{ steps.images.outputs.mig_manager_image }}
+    steps:
+      - uses: actions/checkout@v5
+      
+      - name: Install regctl
+        run: |
+          REGCTL_VERSION=v0.9.2
+          mkdir -p bin
+          curl -sSLo bin/regctl https://github.com/regclient/regclient/releases/download/${REGCTL_VERSION}/regctl-linux-amd64
+          chmod +x bin/regctl
+          echo "$(pwd)/bin" >> $GITHUB_PATH
+      
+      - name: Get latest component images
+        id: images
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Use workflow_dispatch inputs if provided, otherwise fetch latest
```

Comment:
Question: when would someone set these variables as inputs? If the scope of this workflow is to **always** test the top-of-tree images for all operands, then maybe we can remove the input variables and move the entire `run:` block into a shell script.